bindata/network/ovn-kubernetes/self-hosted/ovnkube-master.yaml

```diff
@@ -366,6 +366,7 @@ spec:
                 /usr/share/ovn/scripts/ovn-ctl stop_nb_ovsdb
                 echo "$(date -Iseconds) - nbdb stopped"
                 rm -f /var/run/ovn/ovnnb_db.pid
+                rm -f /var/run/ovn/ovnnbdb-mem-trim-enabled
```

Comment:
This won't cleanup on crash (e.g. `SIGSEGV`), we need to place the enabled file inside to container filesystem so it is removed when the container dies for any reason.

Placing `ovnnb_db.pid` on the host filesystem is a bad special case that we had a different bug for  https://github.com/openshift/cluster-network-operator/pull/1256  

In general we should not be keeping runtime container state on the persistent host file system, or if we really have to, then we have to `rm -f /var/run/ovn/ovnnbdb-mem-trim-enabled` in `postStart`.

Reply:
better place for cleanup is the trap TERM handler

Reply:
TERM won't catch the uncatchable signals.

Reply:
err well I guess only `SIGKILL` and `SIGSTOP` can't be caught.  QE tests with `SIGKILL` now, so after that test memory compaction won't be enabled.

`SIGSEGV` is the main signal we expect and we don't trap it currently, the systemd core dump service catches that.

But again, why are we defeating the container lifecycle by putting a file outside the container?  Stop fighting the container.

Reply:
even the `ovnnb_db.pid` is invalid outside the container since we have a PID namespace.

Reply:
SIGKILL can't be trapped so I think if u are testing with SIGKILL we won't be able to even stop any of the dbs u will probably see another class of issues right ?